doc/reference/modules/basic_mapping.xml

```diff
@@ -3353,24 +3353,25 @@
                 example, for audit logs, user session data, etc.
             </para>
 
-            <programlisting><![CDATA[<any name="AnyEntity" id-type="Int64" meta-type="Eg.Custom.Class2TablenameType">
-    <column name="table_name"/>
+            <programlisting><![CDATA[<any name="AnyEntity" id-type="long" meta-type="class">
+    <column name="type"/>
     <column name="id"/>
 </any>]]></programlisting>
 
             <para>
                 The <literal>meta-type</literal> attribute allows the application to specify a custom type that maps
                 database column values to persistent classes that have identifier properties of the type specified by
-                <literal>id-type</literal>. If the meta-type returns instances of <literal>System.Type</literal>,
-                nothing else is required. On the other hand, if it is a basic type like <literal>String</literal> or
-                <literal>Char</literal>, you must specify the mapping from values to classes.
+                <literal>id-type</literal>. If the meta-type is <literal>class</literal>, nothing else is required.
+                The class full name will be persisted in the database as the type of the associated entity.
+                On the other hand, if it is a basic type like <literal>string</literal> or
+                <literal>char</literal>, you must specify the mapping from values to classes.
```

Comment:
This is not actually true: `string` and any string like types may work without meta-values. In such case they work somewhat similarly to `class`, but without a dedicated NHibernate `MetaType`. That causes issues, like those not fixed in #1803. So I have decide to keep that sentence.

If we fix this, this part would have to be updated.